external/esp_idf_port/esp32/esp32_wifi_utils.c

```diff
@@ -245,13 +246,20 @@ static esp_err_t event_handler(void *ctx, system_event_t *event)
 		s_retry_num++;
 		int reason = event->event_info.disconnected.reason;
 		ESP_LOGI(TAG, "[%s] connect to the AP %s fail %d: %d; %d\n", __func__, event->event_info.disconnected.ssid, s_retry_num, event->event_info.disconnected.reason, isStaConnected);
-    
+        
         /*give chance to other thread to continue*/
         sleep(1);
 		if (s_retry_num > 0) {
 			ESP_LOGI(TAG, "g_cbk.sta_connected fail\n");
-			g_cbk.sta_connected(WIFI_UTILS_FAIL, NULL);
-			s_retry_num = 0;
+            if (s_retry_num < WIFI_CONN_RETRY) {
+                esp_wifi_connect();
+            }
+
+            if (s_retry_num == WIFI_CONN_RETRY) {
+                g_cbk.sta_connected(WIFI_UTILS_FAIL, NULL);
+                s_retry_num = 0;
+            }
+
 		} else {
```

Comment:
Usless code. s_retry_num could not be negative number, it wiill be 0~5.

Reply:
it is signed int, and initial as -1 is used for  to differ case "disconnect ap" from "connect ap fail".